clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -9380,21 +9380,16 @@ void AMDGPUTargetCodeGenInfo::setFunctionDeclAttributes(
   if (ReqdWGS || FlatWGS) {
     unsigned Min = 0;
     unsigned Max = 0;
-    unsigned XDim = 0;
-    unsigned YDim = 0;
-    unsigned ZDim = 0;
-    ASTContext &Ctx = M.getContext();
     if (FlatWGS) {
-      Min = FlatWGS->getMin()->EvaluateKnownConstInt(Ctx).getExtValue();
-      Max = FlatWGS->getMax()->EvaluateKnownConstInt(Ctx).getExtValue();
-    }
-    if (ReqdWGS) {
-      XDim = ReqdWGS->getXDimVal()->getZExtValue();
-      YDim = ReqdWGS->getYDimVal()->getZExtValue();
-      ZDim = ReqdWGS->getZDimVal()->getZExtValue();
+      Min = FlatWGS->getMin()
+                ->EvaluateKnownConstInt(M.getContext())
+                .getExtValue();
+      Max = FlatWGS->getMax()
+                ->EvaluateKnownConstInt(M.getContext())
+                .getExtValue();
     }
     if (ReqdWGS && Min == 0 && Max == 0)
-      Min = Max = XDim * YDim * ZDim;
+      Min = Max = ReqdWGS->getXDim() * ReqdWGS->getYDim() * ReqdWGS->getZDim();
 
     if (Min != 0) {
       assert(Min <= Max && "Min must be less than or equal Max");
```

Comment:
```suggestion
      assert(Min <= Max && "Min must be less than or equal to Max");
```

Reply:
Yes, this is from upstream.

>>I worry changing in intel/llvm unnecessarily increases the risk of conflicts during pulldowns.

If we want to change the wording here, we need to do this in upstream instead of intel/llvm.